### ISSUE TITLE
shuffle in two-pass mode omits newlines before sequence headers with wrapped sequences

### ISSUE THREAD

**Issue by brandoninvergo** [shenwei356/seqkit#209]
### Prerequisites

- [x] make sure you're are using the latest version by `seqkit version`
- [x] read the [usage](http://bioinf.shenwei.me/seqkit/usage/)

### Describe your issue

When I run `seqkit shuffle` (0.16.0) in two-pass mode on some NCBI GenBank genomes, there is no newline between the end of a sequence and the `>` header of the next sequence.

Example:
```
$ rsync -azz rsync://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/079/185/GCA_900079185.1_AG_v1/GCA_900079185.1_AG_v1_genomic.fna.gz .
$ gunzip GCA_900079185.1_AG_v1_genomic.fna.gz
$ grep ">" GCA_900079185.1_AG_v1_genomic.fna | head -n 3
>FKIZ01011899.1 Absidia glauca strain CBS 101.48 genome assembly, contig: contig_11899, whole genome shotgun sequence
>FKIZ01011900.1 Absidia glauca strain CBS 101.48 genome assembly, contig: contig_11900, whole genome shotgun sequence
>LT552727.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_6333, whole genome shotgun sequence
$ seqkit shuffle -2 GCA_900079185.1_AG_v1_genomic.fna | grep ">" | head -n3
[INFO] create and read FASTA index ...
[INFO] create FASTA index for GCA_900079185.1_AG_v1_genomic.fna
[INFO] read sequence IDs from FASTA index ...
[INFO] 5338 sequences loaded
[INFO] shuffle ...
[INFO] output ...
>LT553270.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_7956, whole genome shotgun sequence
gcggtgtTCTTTGTTGATTGGCAAAG>LT552546.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_5961, whole genome shotgun sequence
CAACGAGGTCTCATTTTCCTGCA>LT554062.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_10644, whole genome shotgun sequence
```
Turning off two-pass mode gives the expected results:
```
$ seqkit shuffle GCA_900079185.1_AG_v1_genomic.fna | grep ">" | head -n3
[INFO] read sequences ...
[INFO] 5338 sequences loaded
[INFO] shuffle ...
[INFO] output ...
>LT553270.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_7956, whole genome shotgun sequence
>LT552546.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_5961, whole genome shotgun sequence
>LT554062.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_10644, whole genome shotgun sequence
```

I note that this might be related to #173 and fix 880fe5b. When I run the example in that bug report, I do not observe the present bug:

```
$ grep ">" test.fasta
>f9b7e3d269fe4babaec34f5e196a0d27
>7aad99eff09746dd9c5dc763c11fc43d
>fff1e7d7226f475ebfed96329efa008e
>89571a38a8d34548b44c54987286c97d
>44a8e1f5c2ea4c788ed6a72e3c3dc965
$ seqkit shuffle -2 test.fasta | grep ">"
[INFO] create and read FASTA index ...
[INFO] read sequence IDs from FASTA index ...
[INFO] 5 sequences loaded
[INFO] shuffle ...
[INFO] output ...
>44a8e1f5c2ea4c788ed6a72e3c3dc965
>f9b7e3d269fe4babaec34f5e196a0d27
>fff1e7d7226f475ebfed96329efa008e
>7aad99eff09746dd9c5dc763c11fc43d
>89571a38a8d34548b44c54987286c97d
```
I have identified the issue that when I copied in the example from #173, the sequence lines were not wrapped. The NCBI Genome file is wrapped to 80 characters. If I unwrap them, everything is fine:

```
$ awk 'BEGIN {RS=">";FS="\n";OFS=""} NR>1 {print ">"$1; $1=""; print}' GCA_900079185.1_AG_v1_genomic.fna >test.fa
$ seqkit shuffle -2 test2.fa  | grep ">" | head -n3
[INFO] create and read FASTA index ...
[INFO] create FASTA index for test2.fa
[INFO] read sequence IDs from FASTA index ...
[INFO] 5338 sequences loaded
[INFO] shuffle ...
[INFO] output ...
>LT553270.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_7956, whole genome shotgun sequence
>LT552546.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_5961, whole genome shotgun sequence
>LT554062.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_10644, whole genome shotgun sequence
```

Weirdly, if I try to do it in a pipeline, it fails again:
```
$ awk 'BEGIN {RS=">";FS="\n";OFS=""} NR>1 {print ">"$1; $1=""; print}' GCA_900079185.1_AG_v1_genomic.fna | seqkit shuffle -2 | grep ">" | head -n3
[INFO] read and write sequences to tempory file: stdin.fastx ...
[INFO] 5338 sequences saved
[INFO] create and read FASTA index ...
[INFO] create FASTA index for stdin.fastx
[INFO] read sequence IDs from FASTA index ...
[INFO] 5338 sequences loaded
[INFO] shuffle ...
[INFO] output ...
>LT553270.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_7956, whole genome shotgun sequence
gcggtgtTCTTTGTTGATTGGCAAAG>LT552546.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_5961, whole genome shotgun sequence
CAACGAGGTCTCATTTTCCTGCA>LT554062.1 Absidia glauca strain CBS 101.48 genome assembly, scaffold: scf_10644, whole genome shotgun sequence
```

**Comment by shenwei356** [comment]
Sorry, the "fix" in #173 introduced this bug.

- [seqkit_darwin_amd64.tar.gz](https://github.com/shenwei356/seqkit/files/6463054/seqkit_darwin_amd64.tar.gz)
- [seqkit_linux_amd64.tar.gz](https://github.com/shenwei356/seqkit/files/6463056/seqkit_linux_amd64.tar.gz)

BTW, you can just run `seqkit seq -w 0` to unwrap sequences.

**Comment by brandoninvergo** [comment]
Thanks for the quick fix!

> BTW, you can just run seqkit seq -w 0 to unwrap sequences.

Ah right, good to know, thanks. I'm new to seqkit so I'm still learning all the different sub-commands.
